src/Http/Client.php

```diff
@@ -519,20 +520,26 @@ public function buildUrl(string $url, $query = [], array $options = []): string
         if (empty($options) && empty($query)) {
             return $url;
         }
-        if ($query) {
-            $q = strpos($url, '?') === false ? '?' : '&';
-            $url .= $q;
-            $url .= is_string($query) ? $query : http_build_query($query);
-        }
         $defaults = [
             'host' => null,
             'port' => null,
             'scheme' => 'http',
             'basePath' => '',
             'protocolRelative' => false,
+            'query_encoding' => PHP_QUERY_RFC1738,
```

Comment:
Could this be `queryEncoding`? Do we need an option at all. Seems like we could/should default to 3986 given that 1738 is updated by 3986.

Reply:
No problem, I've changed the name to  queryEncoding.